core/src/main/java/org/apache/calcite/sql/SqlUtil.java

```diff
@@ -422,10 +422,43 @@ public static void unparseBinarySyntax(
     writer.setNeedWhitespace(needsSpace);
     writer.sep(operator.getName());
     writer.setNeedWhitespace(needsSpace);
+    processNode(call.operand(1), call);
     call.operand(1).unparse(writer, operator.getRightPrec(), rightPrec);
     writer.endList(frame);
   }
 
+  private static void processNode(SqlNode binaryNode, SqlCall call) {
+    if (binaryNode instanceof SqlNodeList) {
+      SqlNodeList operandList = (SqlNodeList) binaryNode;
+      processNodeList(operandList);
+    }
+    if (binaryNode instanceof SqlCharStringLiteral && binaryNode.toString().contains("\n")) {
+      binaryNode = replaceNewLineChar(binaryNode);
+      call.setOperand(1, binaryNode);
+    }
+  }
+
+  public static void processNodeList(SqlNodeList operandList) {
+    SqlNode node;
+    for (SqlNode operand : operandList) {
+      if (operand instanceof SqlCharStringLiteral) {
+        node = operand.toString().contains("\\") ? replaceSingleBackslashes(operand) : operand;
+        node = operand.toString().contains("\n") ? replaceNewLineChar(node) : node;
```

Comment:
Here also instead of operand.toString() you can write node.toString()

Reply:
Done